osu.Game/Beatmaps/Formats/LegacyDecoder.cs

```diff
@@ -40,7 +41,53 @@ protected override void ParseStreamInto(StreamReader stream, T beatmap)
 
         protected virtual bool ShouldSkipLine(string line) => string.IsNullOrWhiteSpace(line) || line.StartsWith("//");
 
-        protected abstract void ParseLine(T output, Section section, string line);
+        protected virtual void ParseLine(T output, Section section, string line)
+        {
+            switch (section)
+            {
+                case Section.Colours:
+                    handleColours(output, line);
+                    return;
+            }
+        }
+
+        private bool hasCustomColours;

```

Comment:
I'd rather this was named `hasComboColours` so as to not conflict with `IHasCustomColours`.

Reply:
oops